runtime/src/iree/hal/cts/driver_test.h

```diff
@@ -22,25 +22,124 @@ namespace cts {
+                             device_infos[i].name.size)
+              << "'..\n";
+    iree_hal_device_t* device = NULL;
+    IREE_ASSERT_OK(iree_hal_driver_create_device_by_ordinal(
+        driver_, i, /*param_count=*/0, /*params=*/NULL, iree_allocator_system(),
+        &device));
+    iree_string_view_t device_id = iree_hal_device_id(device);
+    std::cout << "    Created device with id: '"
+              << std::string(device_id.data, device_id.size) << "'\n";
+    iree_hal_device_release(device);
+  }
+
+  iree_allocator_free(iree_allocator_system(), device_infos);
+}
+
+TEST_P(driver_test, QueryAndCreateAvailableDevicesByPath) {
+  iree_host_size_t device_info_count = 0;
+  iree_hal_device_info_t* device_infos = NULL;
+  IREE_ASSERT_OK(iree_hal_driver_query_available_devices(
+      driver_, iree_allocator_system(), &device_info_count, &device_infos));
+
+  std::cout << "Driver has " << device_info_count << " device(s)\n";
+
+  if (device_info_count > 0) {
+    iree_string_view_t name = device_infos[0].name;
+    std::cout << "  Creating device '" << std::string(name.data, name.size)
+              << "' with empty path..\n";
+    iree_hal_device_t* device = NULL;
+    IREE_ASSERT_OK(iree_hal_driver_create_device_by_path(
+        driver_, name, iree_string_view_empty(),
+        /*param_count=*/0, /*params=*/NULL, iree_allocator_system(), &device));
+    iree_string_view_t device_id = iree_hal_device_id(device);
+    std::cout << "    Created device with id: '"
+              << std::string(device_id.data, device_id.size) << "'\n";
+    iree_hal_device_release(device);
+  }
+
+  for (iree_host_size_t i = 0; i < device_info_count; ++i) {
+    iree_string_view_t name = device_infos[i].name;
+    std::cout << "  Creating device '" << std::string(name.data, name.size)
+              << "' with index #" << i << "..\n";
+    iree_hal_device_t* device = NULL;
+    char index[8];
+    IREE_ASSERT_LT(snprintf(index, 8, "%d", i), 8);
+    IREE_ASSERT_OK(iree_hal_driver_create_device_by_path(
+        driver_, name, iree_string_view_literal(index),
+        /*param_count=*/0,
+        /*params=*/NULL, iree_allocator_system(), &device));
+    iree_string_view_t device_id = iree_hal_device_id(device);
+    std::cout << "    Created device with id: '"
+              << std::string(device_id.data, device_id.size) << "'\n";
+    iree_hal_device_release(device);
+  }
+  if (device_info_count > 0) {
+    iree_string_view_t name = device_infos[0].name;
+    std::cout << "  Creating device '" << std::string(name.data, name.size)
+              << "' with index #" << device_info_count << "..\n";
+    iree_hal_device_t* device = NULL;
+    char index[8];
+    IREE_ASSERT_LT(snprintf(index, 8, "%d", device_info_count), 8);
+    iree_status_t status = iree_hal_driver_create_device_by_path(
+        driver_, name, iree_string_view_literal(index),
+        /*param_count=*/0,
+        /*params=*/NULL, iree_allocator_system(), &device);
```

Comment:
```
C:\a\iree\iree\runtime/src/iree/hal/cts/driver_test.h(104): error C7555: use of designated initializers requires at least '/std:c++20'
C:\a\iree\iree\runtime/src/iree/hal/cts/driver_test.h(121): error C7555: use of designated initializers requires at least '/std:c++20'
C:\a\iree\iree\runtime/src/iree/hal/cts/driver_test.h(138): error C7555: use of designated initializers requires at least '/std:c++20'
```
(try one of the functions / macros instead of `iree_string_view_literal`, this is compiled as C++ and not C)

https://github.com/iree-org/iree/blob/e5a8374bb198cbf8a73eb1a2b39d8b9d9ee973d8/runtime/src/iree/base/string_view.h#L90-L97

Reply:
Ha, thanks for the pointers!